core/src/test/java/io/substrait/type/proto/GenericRoundtripTest.java

```diff
@@ -0,0 +1,145 @@
+package io.substrait.type.proto;
+
+import com.google.protobuf.ByteString;
+import io.substrait.expression.Expression;
+import io.substrait.expression.ExpressionCreator;
+import io.substrait.expression.proto.ExpressionProtoConverter;
+import io.substrait.expression.proto.ProtoExpressionConverter;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+public class GenericRoundtripTest {
+    static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(GenericRoundtripTest.class);
+
+    static Random rand = new Random(123);
```

Comment:
Let's do `private final` instead of static. Random is not thread safe so best to avoid accidentally sharing.

Reply:
Also best to add a comment that the value is not meaningful, just done to make sure test is repeatable.

Reply:
I guess you already merged. Should I send in an upate? (Or we can do it in future iterations, as I assume we should expand this to also cover complex types).